File: frontend/types/expense-category.ts
export interface ExpenseCategory {
  id: number;
  name: string;
  in_use: boolean;
}

export interface ExpenseCategoryListResponse {
  items: ExpenseCategory[];
  total: number;
}

export interface ExpenseCategoryCreate {
  name: string;
}

export interface ExpenseCategoryUpdate {
  name: string;
}

File: backend/app/schemas/expense_category.py
"""Expense category schemas."""

from pydantic import BaseModel, Field


class ExpenseCategoryCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)


class ExpenseCategoryUpdate(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)


class ExpenseCategoryResponse(BaseModel):
    id: int
    name: str
    in_use: bool = False

    class Config:
        from_attributes = True


class ExpenseCategoryListResponse(BaseModel):
    items: list[ExpenseCategoryResponse]
    total: int
